Interop/Indexer/src/main/kotlin/org/jetbrains/kotlin/native/interop/indexer/J2ObjCParser.kt

```diff
@@ -100,10 +100,28 @@ class J2ObjCParser: ClassVisitor(Opcodes.ASM5) {
         nsReturnsRetained = false,
         isOptional = false,
         isInit = false,
-        isExplicitlyDesignatedInitializer = false)
+        isExplicitlyDesignatedInitializer = false,
+        j2objcNameOverride = true)
     }
   }
 
+  private fun buildJ2objcMethodName(methodName: String, methodDesc: String): String {
+    val outputMethodName = StringBuilder(methodName)
+
+    val types = getArgumentTypes(methodDesc)
+    if (types.size > 0) {
+      outputMethodName.append("With" + types.get(0).className.capitalize() + ":")
+    }
+    if (types.size > 1) {
+      for (i in 1 until types.size) {
+        outputMethodName.append("with" + types.get(i).className.capitalize() + ":")
+      }
+    }
```

Comment:
Yep. Done.

I could replace this with map, would that be a good idea?

Reply:
I not sure -> It doesn't look like this code has changed to me ...

What I meant is:
    if (types.size > 1) {
      for (i in 1 until types.size) {
        outputMethodName.append("with" + types.get(i).className.capitalize() + ":")
      }
    }

---- could be ----
      for (i in 1 until types.size) {
        outputMethodName.append("with" + types.get(i).className.capitalize() + ":")
      }

because if types.size <= 1 the for loop won't execute.

A map statement may be more readable so it's up to you if you want to make that translation.